apps/www/_blog/2023-11-01-supabase-is-now-compatible-with-nextjs-14.mdx

```diff
@@ -0,0 +1,192 @@
+- [Middleware](https://nextjs.org/docs/app/building-your-application/routing/middleware)
+
+And for the [Pages Router](https://nextjs.org/docs/pages):
+
+- [`getServerSideProps` function](https://nextjs.org/docs/pages/building-your-application/data-fetching/get-server-side-props)
+- [`getStaticProps` function](https://nextjs.org/docs/pages/building-your-application/data-fetching/get-static-props)
+- [API Routes](https://nextjs.org/docs/pages/building-your-application/routing/api-routes)
+- [Page Components](https://nextjs.org/docs/pages/building-your-application/data-fetching/client-side)
+
+## What problem needed to be solved?
+
+By default, [`supabase-js`](https://supabase.com/docs/reference/javascript/introduction) uses `localStorage` to store the user's session. This is great for client-side apps, but it means that when you try to use `supabase-js` in any of the Next.js server bits, it will fail as there is no concept of `localStorage` on the server.
+
+Therefore, we need to configure `supabase-js` to use cookies instead of `localStorage` when running on the server. But this code is a little verbose to ask people to duplicate across every app they build with Supabase:
+
+```tsx
+const supabase = createClient(supabaseUrl, supabaseAnonKey, {
+  auth: {
+    flowType: 'pkce',
+    autoRefreshToken: false,
+    detectSessionInUrl: false,
+    persistSession: true,
+    storage: {
+      getItem: async (key: string) => {
+        cookieStore.get(key)
+      },
+      setItem: async (key: string, value: string) => {
+        cookieStore.set(key, value)
+      },
+      removeItem: async (key: string) => {
+        cookieStore.remove(key)
+      },
+    },
+  },
+})
+```
+
+That takes care of the server-side pieces of Next.js, but since we recommend securing your apps with [Row Level Security (RLS) policies](https://www.youtube.com/watch?v=Ow_Uzedfohk), you can safely access your user's session client-side too. Therefore, we need to tell the browser how access that cookie too:
+
+```tsx
+const supabase = createClient(supabaseUrl, supabaseAnonKey, {
+  auth: {
+    flowType: 'pkce',
+    autoRefreshToken: true,
+    detectSessionInUrl: true,
+    persistSession: true,
+    storage: {
+      getItem: async (key: string) => {
+        return parse(document.cookie[key])
+      },
+      setItem: async (key: string, value: string) => {
+        document.cookie = serialize(key, value)
+      },
+    },
+    removeItem: async (key: string) => {
+      document.cookie = serialize(key, '', {
+        maxAge: 0,
+      })
+    },
+  },
+})
+```
+
+That is a lot of very confusing code! So we decided to [create a package called `@supabase/ssr`](https://supabase.com/docs/guides/auth/server-side/overview) that does all of this for you. Then we took it one step further and created a [Next.js and Supabase Starter Template](https://supabase.com/docs/guides/getting-started/quickstarts/nextjs), so you can just focus on building your awesome app! 🚀
```

Comment:
Maybe not the place, but wonder if we should mention that ssr is a successor to auth-helpers? Better, simpler, easier syntax, easier to understand etc....

since we're getting a bit of flack on twitter

Reply:
This will be its own blog coming soon 👍